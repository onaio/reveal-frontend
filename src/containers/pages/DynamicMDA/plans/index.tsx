import reducerRegistry from '@onaio/redux-reducer-registry';
import React from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { Store } from 'redux';
import {
  HIDDEN_PLAN_STATUSES,
  SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
} from '../../../../configs/env';
import { MDA_PLANS } from '../../../../configs/lang';
import { QUERY_PARAM_TITLE, REPORT_MDA_PLAN_URL } from '../../../../constants';
import { getPlanStatusToDisplay, getQueryParams, RouteParams } from '../../../../helpers/utils';
import supersetFetch from '../../../../services/superset';
import GenericPLansReducer, {
  genericFetchPlans,
  makeGenericPlansArraySelector,
  reducerName as GenericPLansReducerName,
} from '../../../../store/ducks/generic/plans';
import { GenericPlan } from '../../../../store/ducks/generic/plans';
import { InterventionType } from '../../../../store/ducks/plans';
import { GenericPlanListProps, GenericPlansList } from '../../GenericPlansList';

/** register the MDA plan definitions reducer */
reducerRegistry.register(GenericPLansReducerName, GenericPLansReducer);

/** a list of plan statuses to be displayed */
const allowedPlanStatusList = getPlanStatusToDisplay(HIDDEN_PLAN_STATUSES);

/** selector for Dynamic MDA plans */
const MDAPlansArraySelector = makeGenericPlansArraySelector();

/** Simple component that loads a list of MDA plans */
const MDAPLansList = (props: GenericPlanListProps & RouteComponentProps) => {
  return (
    <div>
      <GenericPlansList {...props} />
    </div>
  );
};

/** Declare default props for MDAPLansList */
const defaultProps: GenericPlanListProps = {
  fetchPlans: genericFetchPlans,
  pageTitle: MDA_PLANS,
  pageUrl: REPORT_MDA_PLAN_URL,
  plans: [],
  service: supersetFetch,
  supersetReportingSlice: SUPERSET_DYNAMIC_MDA_REPORTING_PLANS_SLICE,
};

MDAPLansList.defaultProps = defaultProps;

export { MDAPLansList };

/** Connect the component to the store */

/** interface to describe props from mapStateToProps */
interface DispatchedStateProps extends RouteComponentProps<RouteParams> {
  plans: GenericPlan[];
}

/** map state to props */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: RouteComponentProps<RouteParams>
): DispatchedStateProps => {
  const searchedTitle = getQueryParams(ownProps.location)[QUERY_PARAM_TITLE] as string;
  const plans = MDAPlansArraySelector(state, {
    interventionTypes: [InterventionType.DynamicMDA],
    plan_title: searchedTitle,
    statusList: allowedPlanStatusList,
  });

  return {
    ...ownProps,
    plans,
  };
};

/** map dispatch to props */
const mapDispatchToProps = { fetchPlans: genericFetchPlans };

/** Connected MDA Plans component */
const ConnectedMDAPLansList = connect(mapStateToProps, mapDispatchToProps)(MDAPLansList);

export default ConnectedMDAPLansList;
