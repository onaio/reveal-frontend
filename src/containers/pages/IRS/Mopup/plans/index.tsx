import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { MOP_UP_REPORTING_TITLE } from '../../../../../configs/lang';
import { IRS_MOP_UP_REPORT_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSMopUpReporting = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: MOP_UP_REPORTING_TITLE,
    pageUrl: IRS_MOP_UP_REPORT_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
