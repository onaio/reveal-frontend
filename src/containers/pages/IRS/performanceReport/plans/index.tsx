import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { IRS_PERFORMANCE_REPORTING_TITLE } from '../../../../../configs/lang';
import { PERFORMANCE_REPORT_IRS_PLAN_URL } from '../../../../../constants';
import { RouteParams } from '../../../../../helpers/utils';
import ConnectedIRSPlansList from '../../plans';

export const IRSPerfomenceReport = (props: RouteComponentProps<RouteParams>) => {
  const plansProps = {
    ...props,
    pageTitle: IRS_PERFORMANCE_REPORTING_TITLE,
    pageUrl: PERFORMANCE_REPORT_IRS_PLAN_URL,
  };
  return <ConnectedIRSPlansList {...plansProps} />;
};
