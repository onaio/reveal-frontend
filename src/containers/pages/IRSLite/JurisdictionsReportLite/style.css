.irs-report-table .ReactTable .rt-thead.-headerGroups .rt-th {
  white-space: normal;
}

.irs-report-table .ReactTable .rt-thead.-header .rt-th {
  font-weight: normal;
  white-space: normal;
}

.irs-report-indicator-container {
  padding: 7px 5px;
}

.irs-report-table .ReactTable .rt-resizable-header {
  -webkit-box-shadow: none !important;
  box-shadow: none !important;
}

.irs-report-table .ReactTable .rt-resizer:before {
  display: inline-block;
  position: absolute;
  right: 25px;
  top: 3px;
  color: transparent;
  content: '.';
  background-size: 18px 18px;
  background-repeat: no-repeat;
  height: 0;
  width: 0;
}

.irs-report-table .ReactTable .rt-th:hover .rt-resizer:before {
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid #999;
}

.irs-report-table .ReactTable .rt-th.-sort-asc .rt-resizer:before,
.irs-report-table .ReactTable .rt-th.-sort-asc:hover .rt-resizer:before {
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-bottom: 5px solid black;
}

.irs-report-table .ReactTable .rt-th.-sort-desc .rt-resizer:before,
.irs-report-table .ReactTable .rt-th.-sort-desc:hover .rt-resizer:before {
  border-left: 5px solid transparent;
  border-right: 5px solid transparent;
  border-top: 5px solid black;
}
