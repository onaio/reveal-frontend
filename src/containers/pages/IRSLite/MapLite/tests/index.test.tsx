import reducerRegistry from '@onaio/redux-reducer-registry';
import superset from '@onaio/superset-connector';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Helmet } from 'react-helmet';
import { Router } from 'react-router';
import { IRSLiteReportingMap } from '../';
import { GisidaLiteProps } from '../../../../../components/GisidaLite';
import { MAP, REPORT_IRS_LITE_PLAN_URL } from '../../../../../constants';
import store from '../../../../../store';
import GenericJurisdictionsReducer, {
  fetchGenericJurisdictions,
  getGenericJurisdictionByJurisdictionId,
  reducerName as GenericJurisdictionsReducerName,
} from '../../../../../store/ducks/generic/jurisdictions';
import IRSPlansReducer, {
  GenericPlan,
  reducerName as IRSPlansReducerName,
} from '../../../../../store/ducks/generic/plans';
import genericStructuresReducer, {
  fetchGenericStructures,
  getGenericStructures,
  reducerName as genericStructuresReducerName,
  StructureFeatureCollection,
} from '../../../../../store/ducks/generic/structures';
import jurisdictionReducer, {
  fetchJurisdictions,
  getJurisdictionById,
  reducerName as jurisdictionReducerName,
} from '../../../../../store/ducks/jurisdictions';
import { InterventionType } from '../../../../../store/ducks/plans';
import * as fixtures from '../../JurisdictionsReportLite/fixtures';

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

/** register the reducers */
reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
reducerRegistry.register(jurisdictionReducerName, jurisdictionReducer);
reducerRegistry.register(GenericJurisdictionsReducerName, GenericJurisdictionsReducer);
reducerRegistry.register(genericStructuresReducerName, genericStructuresReducer);

/** set up data in the store needed for this view */
const focusAreaData = superset.processData(fixtures.ZambiaFocusAreasJSON) || [];
const structureData = superset.processData(fixtures.ZambiaStructuresJSON) || [];
const jurisdictionData = superset.processData(fixtures.ZambiaJurisdictionsJSON) || [];
const jurisdictionData2 = superset.processData(fixtures.ZambiaJurisdictionsGeojsonJSON) || [];

store.dispatch(fetchGenericJurisdictions('zm-focusAreas', focusAreaData));
store.dispatch(fetchGenericStructures('zm-structures', structureData));
store.dispatch(fetchJurisdictions(jurisdictionData));
store.dispatch(fetchJurisdictions(jurisdictionData2));

const history = createBrowserHistory();

export const plans = [
  {
    jurisdiction_root_parent_ids: ['f45b9380-c970-4dd1-8533-9e95ab12f128'],
    plan_date: '2019-09-05',
    plan_effective_period_end: '2019-09-25',
    plan_effective_period_start: '2019-09-05',
    plan_id: '17f89152-51a6-476c-9246-8fee6f9e6ebf',
    plan_intervention_type: InterventionType.IRSLite,
    plan_name: 'IRS-2019-09-05-TEST',
    plan_status: 'retired',
    plan_title: 'IRS 2019-09-05 TEST',
    plan_version: '2',
  },
];

jest.mock('../../../../../configs/env', () => ({
  GISIDA_MAPBOX_TOKEN: 'hunter2',
  GISIDA_TIMEOUT: 3000,
  HIDDEN_MAP_LEGEND_ITEMS: [],
  SUPERSET_IRS_LITE_REPORTING_INDICATOR_ROWS: 'zambia2020',
  SUPERSET_IRS_LITE_REPORTING_INDICATOR_STOPS: 'zambia2020',
  SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_DATA_SLICES: '11',
  SUPERSET_IRS_LITE_REPORTING_JURISDICTIONS_FOCUS_AREA_SLICE: '12',
  SUPERSET_IRS_LITE_REPORTING_PLANS_SLICE: '13',
  SUPERSET_IRS_LITE_REPORTING_STRUCTURES_DATA_SLICE: '14',
  SUPERSET_JURISDICTIONS_SLICE: 1,
  SUPERSET_MAX_RECORDS: 2000,
}));

jest.mock('../../../../../components/GisidaLite', () => {
  const MemoizedGisidaLiteMock = () => <div>Mock component</div>;
  return {
    MemoizedGisidaLite: MemoizedGisidaLiteMock,
  };
});

describe('components/IRS Reports/IRSLiteReportingMap', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();
    const props = {
      focusArea: getGenericJurisdictionByJurisdictionId(
        store.getState(),
        'zm-focusAreas',
        'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
      ),
      history,
      jurisdiction: getJurisdictionById(store.getState(), 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'),
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdictionId: 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2',
          planId: (plans[0] as GenericPlan).plan_id,
        },
        path: `${REPORT_IRS_LITE_PLAN_URL}/:planId/:jurisdictionId/${MAP}`,
        url: `${REPORT_IRS_LITE_PLAN_URL}/${
          (plans[0] as GenericPlan).plan_id
        }/ce13e7f4-6926-4be0-9117-519bd1cc4bb2/${MAP}`,
      },
      plan: plans[0] as GenericPlan,
      structures: getGenericStructures(
        store.getState(),
        'zm-structures',
        'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
      ),
    };
    shallow(
      <Router history={history}>
        <IRSLiteReportingMap {...props} />
      </Router>
    );
  });

  it('renders without structures', () => {
    const mock: any = jest.fn();
    const props = {
      focusArea: getGenericJurisdictionByJurisdictionId(
        store.getState(),
        'zm-focusAreas',
        'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
      ),
      history,
      jurisdiction: getJurisdictionById(store.getState(), 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'),
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdictionId: 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2',
          planId: (plans[0] as GenericPlan).plan_id,
        },
        path: `${REPORT_IRS_LITE_PLAN_URL}/:planId/:jurisdictionId/${MAP}`,
        url: `${REPORT_IRS_LITE_PLAN_URL}/${
          (plans[0] as GenericPlan).plan_id
        }/ce13e7f4-6926-4be0-9117-519bd1cc4bb2/${MAP}`,
      },
      plan: plans[0] as GenericPlan,
      structures: {} as StructureFeatureCollection,
    };
    shallow(
      <Router history={history}>
        <IRSLiteReportingMap {...props} />
      </Router>
    );
  });

  it('renders correctly', async () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const mock: any = jest.fn();

    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementation(async () => []);

    const focusArea = getGenericJurisdictionByJurisdictionId(
      store.getState(),
      'zm-focusAreas',
      'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
    );

    const structures = getGenericStructures(
      store.getState(),
      'zm-structures',
      'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
    );

    const jurisdiction = getJurisdictionById(
      store.getState(),
      'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
    );

    const props = {
      focusArea,
      history,
      jurisdiction,
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdictionId: 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2',
          planId: (plans[0] as GenericPlan).plan_id,
        },
        path: `${REPORT_IRS_LITE_PLAN_URL}/:planId/:jurisdictionId/${MAP}`,
        url: `${REPORT_IRS_LITE_PLAN_URL}/${
          (plans[0] as GenericPlan).plan_id
        }/ce13e7f4-6926-4be0-9117-519bd1cc4bb2/${MAP}`,
      },
      plan: plans[0] as GenericPlan,
      service: supersetServiceMock,
      structures,
    };
    const wrapper = mount(
      <Router history={history}>
        <IRSLiteReportingMap {...props} />
      </Router>
    );
    const helmet = Helmet.peek();
    await flushPromises();
    expect(toJson(wrapper.find('BreadcrumbItem li'))).toMatchSnapshot('breadcrumbs');
    expect(toJson(wrapper.find('h3.page-title'))).toMatchSnapshot('page title');
    expect(helmet.title).toEqual(
      'IRS 2019-09-05 TEST: so_Sompani_Health_Post_MACEPA_IRS_Lite_2020'
    );
    expect(toJson(wrapper.find('.mapSidebar h5'))).toMatchSnapshot('Sidebar title');
    expect(toJson(wrapper.find('.sidebar-legend-item'))).toMatchSnapshot('Legend items');
    expect(toJson(wrapper.find('.responseItem h6'))).toMatchSnapshot('Response item titles');
    expect(toJson(wrapper.find('.responseItem p.indicator-description'))).toMatchSnapshot(
      'Response item descriptions'
    );
    expect(toJson(wrapper.find('p.indicator-breakdown'))).toMatchSnapshot(
      'Indicator item breakdown'
    );
    expect(toJson(wrapper.find('.responseItem ProgressBar'))).toMatchSnapshot(
      'Response item ProgressBar'
    );

    expect(supersetServiceMock.mock.calls).toEqual([
      [
        1,
        {
          adhoc_filters: [
            {
              clause: 'WHERE',
              comparator: 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'jurisdiction_id',
            },
          ],
          row_limit: 1,
        },
      ],
      [
        '14',
        {
          adhoc_filters: [
            {
              clause: 'WHERE',
              comparator: 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'jurisdiction_id',
            },
            {
              clause: 'WHERE',
              comparator: '17f89152-51a6-476c-9246-8fee6f9e6ebf',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'plan_id',
            },
          ],
          row_limit: 2000,
        },
      ],
      [
        '12',
        {
          adhoc_filters: [
            {
              clause: 'WHERE',
              comparator: 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'jurisdiction_id',
            },
            {
              clause: 'WHERE',
              comparator: '17f89152-51a6-476c-9246-8fee6f9e6ebf',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'plan_id',
            },
          ],
          row_limit: 1,
        },
      ],
      [
        '13',
        {
          adhoc_filters: [
            {
              clause: 'WHERE',
              comparator: '17f89152-51a6-476c-9246-8fee6f9e6ebf',
              expressionType: 'SIMPLE',
              operator: '==',
              subject: 'plan_id',
            },
          ],
          row_limit: 1,
        },
      ],
    ]);
    expect(supersetServiceMock).toHaveBeenCalledTimes(4);
    wrapper.unmount();
  });

  it('renders not found', () => {
    const mock: any = jest.fn();
    const props = {
      focusArea: getGenericJurisdictionByJurisdictionId(
        store.getState(),
        'zm-focusAreas',
        'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
      ),
      history,
      jurisdiction: getJurisdictionById(store.getState(), 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'),
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdictionId: 'bad-id',
          planId: (plans[0] as GenericPlan).plan_id,
        },
        path: `${REPORT_IRS_LITE_PLAN_URL}/:planId/:jurisdictionId/${MAP}`,
        url: `${REPORT_IRS_LITE_PLAN_URL}/${
          (plans[0] as GenericPlan).plan_id
        }/ce13e7f4-6926-4be0-9117-519bd1cc4bb2/${MAP}`,
      },
      plan: plans[0] as GenericPlan,
      structures: {} as StructureFeatureCollection,
    };

    const wrapper = mount(
      <Router history={history}>
        <IRSLiteReportingMap {...props} />
      </Router>
    );

    expect(wrapper.find('NotFound')).not.toBe({});
  });

  it('map props are generated without structures', async () => {
    fetch.mockResponseOnce(JSON.stringify({}));
    const mock: any = jest.fn();

    const supersetServiceMock: any = jest.fn();
    supersetServiceMock.mockImplementation(async () => []);

    const focusArea = getGenericJurisdictionByJurisdictionId(
      store.getState(),
      'zm-focusAreas',
      'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
    );

    const jurisdiction = getJurisdictionById(
      store.getState(),
      'ce13e7f4-6926-4be0-9117-519bd1cc4bb2'
    );

    const props = {
      focusArea,
      history,
      jurisdiction,
      location: mock,
      match: {
        isExact: true,
        params: {
          jurisdictionId: 'ce13e7f4-6926-4be0-9117-519bd1cc4bb2',
          planId: (plans[0] as GenericPlan).plan_id,
        },
        path: `${REPORT_IRS_LITE_PLAN_URL}/:planId/:jurisdictionId/${MAP}`,
        url: `${REPORT_IRS_LITE_PLAN_URL}/${
          (plans[0] as GenericPlan).plan_id
        }/ce13e7f4-6926-4be0-9117-519bd1cc4bb2/${MAP}`,
      },
      plan: plans[0] as GenericPlan,
      service: supersetServiceMock,
      structures: null,
    };
    const wrapper = mount(
      <Router history={history}>
        <IRSLiteReportingMap {...props} />
      </Router>
    );
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();
    const helmet = Helmet.peek();
    expect(helmet.title).toEqual(
      'IRS 2019-09-05 TEST: so_Sompani_Health_Post_MACEPA_IRS_Lite_2020'
    );
    // map zoom, mapCenter and Bounds are generated without structures
    expect(wrapper.find('MemoizedGisidaLiteMock').length).toEqual(1);
    expect((wrapper.find('MemoizedGisidaLiteMock').props() as GisidaLiteProps).zoom).toEqual(11);
    expect((wrapper.find('MemoizedGisidaLiteMock').props() as GisidaLiteProps).mapCenter).toEqual([
      27.669583969324798,
      -16.7185900742225,
    ]);
    expect((wrapper.find('MemoizedGisidaLiteMock').props() as GisidaLiteProps).mapBounds).toEqual([
      27.5511379390647,
      -16.7924169179064,
      27.7880299995849,
      -16.6447632305386,
    ]);
  });
});
