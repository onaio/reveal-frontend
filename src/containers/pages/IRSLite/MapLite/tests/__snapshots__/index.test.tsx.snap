// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: Indicator item breakdown 1`] = `
Array [
  <p
    className="indicator-breakdown"
  >
    Progress: 
    24 of 0 structure(s)
     
    (0%)
  </p>,
  <p
    className="indicator-breakdown"
  >
    Progress: 
    24 of 0 structure(s)
     
    (0%)
  </p>,
  <p
    className="indicator-breakdown"
  >
    Progress: 
    34 of 0 structure(s)
     
    (0%)
  </p>,
  <p
    className="indicator-breakdown"
  >
    Progress: 
    24 of 34 structure(s)
     
    (71%)
  </p>,
]
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: Legend items 1`] = `
Array [
  <div
    className="sidebar-legend-item"
    key="0"
  >
    <span
      className="sidebar-legend-color"
      style={
        Object {
          "backgroundColor": "#68BB0C",
        }
      }
    />
    <span
      className="sidebar-legend-label"
    >
      Sprayed
    </span>
  </div>,
  <div
    className="sidebar-legend-item"
    key="1"
  >
    <span
      className="sidebar-legend-color"
      style={
        Object {
          "backgroundColor": "#F12525",
        }
      }
    />
    <span
      className="sidebar-legend-label"
    >
      Not Sprayed
    </span>
  </div>,
  <div
    className="sidebar-legend-item"
    key="2"
  >
    <span
      className="sidebar-legend-color"
      style={
        Object {
          "backgroundColor": "#FFCA16",
        }
      }
    />
    <span
      className="sidebar-legend-label"
    >
      Not Visited
    </span>
  </div>,
]
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: Response item ProgressBar 1`] = `
Array [
  <ProgressBar
    animate={false}
    cssClass="progress-bar-striped"
    decimalPoints={0}
    height="10px"
    lineColor="#2ECC40"
    lineColorThresholds={
      Object {
        "GREEN_THRESHOLD": Object {
          "color": "#2ECC40",
          "name": "Green",
          "orEquals": true,
          "value": 1,
        },
        "GREY_THRESHOLD": Object {
          "color": "#dddddd",
          "name": "Grey",
          "value": 0.2,
        },
        "RED_THRESHOLD": Object {
          "color": "#FF4136",
          "name": "Red",
          "orEquals": true,
          "value": 0.75,
        },
        "YELLOW_THRESHOLD": Object {
          "color": "#FFDC00",
          "name": "Yellow",
          "value": 0.9,
        },
      }
    }
    max={100}
    min={0}
    showLabel={false}
    stripped={false}
    value={0}
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="progress-bar 
          "
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#dddddd",
            "width": "0%",
          }
        }
      />
    </div>
  </ProgressBar>,
  <ProgressBar
    animate={false}
    cssClass="progress-bar-striped"
    decimalPoints={0}
    height="10px"
    lineColor="#2ECC40"
    lineColorThresholds={
      Object {
        "GREEN_THRESHOLD": Object {
          "color": "#2ECC40",
          "name": "Green",
          "orEquals": true,
          "value": 1,
        },
        "GREY_THRESHOLD": Object {
          "color": "#dddddd",
          "name": "Grey",
          "value": 0.2,
        },
        "RED_THRESHOLD": Object {
          "color": "#FF4136",
          "name": "Red",
          "orEquals": true,
          "value": 0.75,
        },
        "YELLOW_THRESHOLD": Object {
          "color": "#FFDC00",
          "name": "Yellow",
          "value": 0.9,
        },
      }
    }
    max={100}
    min={0}
    showLabel={false}
    stripped={false}
    value={0}
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="progress-bar 
          "
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#dddddd",
            "width": "0%",
          }
        }
      />
    </div>
  </ProgressBar>,
  <ProgressBar
    animate={false}
    cssClass="progress-bar-striped"
    decimalPoints={0}
    height="10px"
    lineColor="#2ECC40"
    lineColorThresholds={
      Object {
        "GREEN_THRESHOLD": Object {
          "color": "#2ECC40",
          "name": "Green",
          "orEquals": true,
          "value": 1,
        },
        "GREY_THRESHOLD": Object {
          "color": "#dddddd",
          "name": "Grey",
          "value": 0.2,
        },
        "RED_THRESHOLD": Object {
          "color": "#FF4136",
          "name": "Red",
          "orEquals": true,
          "value": 0.75,
        },
        "YELLOW_THRESHOLD": Object {
          "color": "#FFDC00",
          "name": "Yellow",
          "value": 0.9,
        },
      }
    }
    max={100}
    min={0}
    showLabel={false}
    stripped={false}
    value={0}
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={0}
        className="progress-bar 
          "
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#dddddd",
            "width": "0%",
          }
        }
      />
    </div>
  </ProgressBar>,
  <ProgressBar
    animate={false}
    cssClass="progress-bar-striped"
    decimalPoints={0}
    height="10px"
    lineColor="#2ECC40"
    lineColorThresholds={
      Object {
        "GREEN_THRESHOLD": Object {
          "color": "#2ECC40",
          "name": "Green",
          "orEquals": true,
          "value": 1,
        },
        "GREY_THRESHOLD": Object {
          "color": "#dddddd",
          "name": "Grey",
          "value": 0.2,
        },
        "RED_THRESHOLD": Object {
          "color": "#FF4136",
          "name": "Red",
          "orEquals": true,
          "value": 0.75,
        },
        "YELLOW_THRESHOLD": Object {
          "color": "#FFDC00",
          "name": "Yellow",
          "value": 0.9,
        },
      }
    }
    max={100}
    min={0}
    showLabel={false}
    stripped={false}
    value={71}
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={71}
        className="progress-bar 
          "
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#FF4136",
            "width": "71%",
          }
        }
      />
    </div>
  </ProgressBar>,
]
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: Response item descriptions 1`] = `
Array [
  <p
    className="indicator-description"
  >
    Percent of Total Structures Sprayed
  </p>,
  <p
    className="indicator-description"
  >
    Percent of Targeted Structures Sprayed
  </p>,
  <p
    className="indicator-description"
  >
    Percent of found structures over targeted
  </p>,
  <p
    className="indicator-description"
  >
    Percent of structures sprayed over found
  </p>,
]
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: Response item titles 1`] = `
Array [
  <h6>
    Spray Coverage (Effectiveness)
  </h6>,
  <h6>
    Targeted Spray Coverage (Effectiveness)
  </h6>,
  <h6>
    Found Coverage
  </h6>,
  <h6>
    Spray Success Rate (PMI SC)
  </h6>,
]
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: Sidebar title 1`] = `
<h5>
  so_Sompani_Health_Post_MACEPA_IRS_Lite_2020
</h5>
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: breadcrumbs 1`] = `
Array [
  <li
    className="breadcrumb-item"
  >
    <Link
      key="0"
      to="/"
    >
      <LinkAnchor
        href="/"
        navigate={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="1"
      to="/intervention/irs-lite/report"
    >
      <LinkAnchor
        href="/intervention/irs-lite/report"
        navigate={[Function]}
      >
        <a
          href="/intervention/irs-lite/report"
          onClick={[Function]}
        >
          IRS Lite Reporting
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="2"
      to="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf"
    >
      <LinkAnchor
        href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf"
        navigate={[Function]}
      >
        <a
          href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf"
          onClick={[Function]}
        >
          IRS 2019-09-05 TEST
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="3"
      to="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/03557b7e-0ddf-41f7-93c8-155669757a16"
    >
      <LinkAnchor
        href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/03557b7e-0ddf-41f7-93c8-155669757a16"
        navigate={[Function]}
      >
        <a
          href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/03557b7e-0ddf-41f7-93c8-155669757a16"
          onClick={[Function]}
        >
          Zambia MACEPA IRS Lite 2020
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="4"
      to="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/48a04a67-ac6f-4d5e-bbc6-09b77ba1253d"
    >
      <LinkAnchor
        href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/48a04a67-ac6f-4d5e-bbc6-09b77ba1253d"
        navigate={[Function]}
      >
        <a
          href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/48a04a67-ac6f-4d5e-bbc6-09b77ba1253d"
          onClick={[Function]}
        >
          Southern MACEPA IRS Lite 2020
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    className="breadcrumb-item"
  >
    <Link
      key="5"
      to="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/032a9542-dab7-4a76-9e2e-bc7eb99a259c"
    >
      <LinkAnchor
        href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/032a9542-dab7-4a76-9e2e-bc7eb99a259c"
        navigate={[Function]}
      >
        <a
          href="/intervention/irs-lite/report/17f89152-51a6-476c-9246-8fee6f9e6ebf/032a9542-dab7-4a76-9e2e-bc7eb99a259c"
          onClick={[Function]}
        >
          Gwembe MACEPA IRS Lite 2020
        </a>
      </LinkAnchor>
    </Link>
  </li>,
  <li
    aria-current="page"
    className="active breadcrumb-item"
  >
    so_Sompani_Health_Post_MACEPA_IRS_Lite_2020
  </li>,
]
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders correctly: page title 1`] = `
<h3
  className="mb-3 page-title"
>
  IRS 2019-09-05 TEST: so_Sompani_Health_Post_MACEPA_IRS_Lite_2020
</h3>
`;

exports[`components/IRS Reports/IRSLiteReportingMap renders not found: Not Found 1`] = `
<NotFound>
  <div>
    404!
  </div>
</NotFound>
`;
